modules/targets/darwin/fonts.nix

```
@@ -10,18 +10,24 @@ let
     pathsToLink = "/share/fonts";
   };
   fonts = "${fontsEnv}/share/fonts";
+  installDir = "${homeDir}/Library/Fonts/HomeManager";
+  versionFile = "${installDir}/derivation";
 in {
   # macOS won't recognize symlinked fonts
   config = mkIf pkgs.stdenv.hostPlatform.isDarwin {
     home.activation.copyFonts = hm.dag.entryAfter [ "writeBoundary" ] ''
       copyFonts() {
-        rm -rf ${homeDir}/Library/Fonts/HomeManager || :
+        if [[ $(cat ${versionFile} 2>/dev/null) = ${fontsEnv} ]]; then
+          _i "No change in font derivation so not touching font files"
+          return 0
+        fi
 
-        local f
-        find -L "${fonts}" -type f -printf '%P\0' | while IFS= read -rd "" f; do
-          $DRY_RUN_CMD install $VERBOSE_ARG -Dm644 -T \
-            "${fonts}/$f" "${homeDir}/Library/Fonts/HomeManager/$f"
-        done
+        $DRY_RUN_CMD mkdir -p ${installDir}
+        $DRY_RUN_CMD ${pkgs.rsync}/bin/rsync $VERBOSE_ARG -acL --chmod=u+w --delete ${fonts}/ ${installDir}
+
+        if [[ ! -v DRY_RUN ]]; then
+          echo ${fontsEnv} > ${versionFile}
+        fi
```

```suggestion
        $DRY_RUN_CMD echo ${escapeShellArg fontsEnv} > ${versionFile}
```
`fontsEnv` declaration could also be updated to use `escapeShellArg` instead.